### DISCUSSION TITLE
Question about the `embed_positions.weight` in Whisper model

### DISCUSSION THREAD

**Discussion by bobqianic** [ggml-org/whisper.cpp#2287] [Q&A]
In the encoder of Whisper models, the `encoder.embed_positions.weight` was initialized using sine and cosine functions of different frequencies. However, in the decoder of Whisper models, the `decoder.embed_positions.weight` consists of trainable parameters that are initialized using random noise. But why is this the case?

| ![encoder embed_positions weight](https://github.com/ggerganov/whisper.cpp/assets/129547291/8f833fac-7d54-459b-939c-a37ddf4fb8cc) |
|:--:|
***encoder.embed_positions.weight***

| ![decoder embed_positions weight](https://github.com/ggerganov/whisper.cpp/assets/129547291/9e8324ca-15e4-4265-9415-cbf5df40a2a3) |
|:--:|
***decoder.embed_positions.weight***

**Comment by bobqianic** [comment]
Maybe...

<img width="527" alt="image" src="https://github.com/ggerganov/whisper.cpp/assets/129547291/b46e2f5b-d4cc-49d2-8538-1d4030f160f9">

**Reply by ggerganov** [reply]
Haha yes, it's voodoo magic
